Vagrantfile

```diff
@@ -1,83 +1,81 @@
+    cluster.vbguest.auto_update = false
   end
-  config.vm.hostname = "node-1"
-  config.vm.network :private_network, ip: "10.42.0.6"
-end
 
-cluster.vm.define "node-2" do |config|
-  config.vm.box = "centos/7"
-  config.ssh.insert_key = false
-  config.vm.provider :virtualbox do |vb, override|
-    vb.customize ["modifyvm", :id, "--memory", "256"]
-    vb.customize ["modifyvm", :id, "--cpus", "1"]
-  end
-  config.vm.hostname = "node-2"
-  config.vm.network :private_network, ip: "10.42.0.7"
-end
+  # For convenience, testing and instruction, all you need is 'vagrant up'
+  # Every vagrant box comes with a user 'vagrant' with password 'vagrant'
+  # Every vagrant box has the root password 'vagrant'
 
-cluster.vm.define "node-3" do |config|
-  config.vm.box = "centos/7"
-  #config.vm.box = "ubuntu/trusty64"
-  config.ssh.insert_key = false
-  config.vm.provider :virtualbox do |vb, override|
-    vb.customize ["modifyvm", :id, "--memory", "256"]
-    vb.customize ["modifyvm", :id, "--cpus", "1"]
-  end
-  config.vm.hostname = "node-3"
-  config.vm.network :private_network, ip: "10.42.0.8"
-end
+  # This vagrant box is downloaded from https://vagrantcloud.com/centos/7
+  # Other variants https://app.vagrantup.com/boxes/search
+  cluster.vm.box = "centos/7"
+  cluster.ssh.insert_key = false
+  # Don't install your own key (you might not have it)
+  # Use this: $HOME/.vagrant.d/insecure_private_key
 
-cluster.vm.define "node-4" do |config|
-  config.vm.box = "centos/7"
-  config.ssh.insert_key = false
-  config.vm.provider :virtualbox do |vb, override|
-    vb.customize ["modifyvm", :id, "--memory", "256"]
-    vb.customize ["modifyvm", :id, "--cpus", "1"]
+  # host to run ansible-core
+  cluster.vm.define "ansible-node", primary: true do |config|
+    config.vm.hostname = "control"
+    config.vm.network :private_network, ip: "10.42.0.2"
+    config.ssh.forward_agent = true
+    config.vm.provider :virtualbox do |vb, override|
+      vb.customize [
+        "modifyvm", :id,
+        "--name", "control",
+        "--memory", "128",
+        "--cpus", 1
+      ]
+    end
   end
-  config.vm.hostname = "node-4"
-  config.vm.network :private_network, ip: "10.42.0.100"
-end
 
-
-cluster.vm.define "tower" do |config|
-  config.vm.box = "ansible/tower"
-  config.ssh.insert_key = false
-  config.vm.provider :virtualbox do |vb, override|
-    vb.customize ["modifyvm", :id, "--memory", "2048"]
-    vb.customize ["modifyvm", :id, "--cpus", "1"]
+  # host to run ansible tower
+  cluster.vm.define "tower", autostart: false do |config|
+    config.vm.box = "ansible/tower"
+    config.vm.hostname = "tower"
+    config.vm.network :private_network, ip: "10.42.0.3"
+    config.vm.provider :virtualbox do |vb, override|
+      vb.customize [
+        "modifyvm", :id,
+        "--name", "control",
+        "--memory", "2048",
+        "--cpus", 1
+      ]
+    end
   end
-  config.vm.hostname = "tower"
-end
 
+  # hosts to run ansible-core
+  (4..7).each do |i|
```

Comment:
Why are we changing the node numbering? Is it *really* necessary? It diverges with how the cloud/AWS provisioner works and I just don't get why we'd start at node-4.

Reply:
The initial number aligns with the IP address. Not a big deal for me. Request changes if it matters.

Reply:
The initial loop index also creates ip numbers in the subnet that comply with the host and other guests in the private network.